### PR TITLE
feat(pd): a gRPC proxy service for tendermint RPCs

### PR DESCRIPTION
**Pull request by erwanor**
This _draft_ PR implements #1232

### CODE REVIEW

File: pd/src/proxy/tendermint.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        // Enquiring with tendermint-rs devs, there was no clean way to do this in the code
+        // I have reviewed. Might be necessary to either:
+        //      - implement `TryFrom` / `TryInto` upstream
+        //      - get the upstream repo to consolidate types
+
+        // Ok(tonic::Response::new(resp.into()));
+        unimplemented!()
+    }
+
+    #[doc = " GetSyncing queries node syncing."]
```
Comment (hdevalence) [inline pd/src/proxy/tendermint.rs:46]: Where'd this `doc` attribute come from? Is this pulling comments from the protos somehow?

File: proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+import "tendermint/types/types.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "cosmos/base/tendermint/v1beta1/types.proto";
+import "cosmos_proto/cosmos.proto";
+import "tendermint/types/block.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/client/grpc/tmservice";
+
+// Service defines the gRPC querier service for tendermint queries.
+// TODO(erwan): rename this back to Service, investigate the ramifications of not doing so
```
Comment (hdevalence) [inline proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto:17]: What was the reason to rename it `TendermintQuery`? name conflicts with `tower::Service` or something?

File: proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+service TendermintQuery {
+  // GetNodeInfo queries the current node info.
+  rpc GetNodeInfo(GetNodeInfoRequest) returns (GetNodeInfoResponse) {
+    option (google.api.http).get = "/cosmos/base/tendermint/v1beta1/node_info";
```
Comment (hdevalence) [inline proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto:21]: No idea what the right answer is here, but should we be annotating these with `cosmos` paths, if we're not doing a cosmos-sdk thing, and thinking of this proto file as something we want to upstream? Or should we just call them `tendermint/blah`?

File: proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto (hunk 2 above)
Comment (erwanor) [reply proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto:17]: Yes, this is just to reduce overhead during development

File: pd/src/proxy/tendermint.rs (hunk 1 above)
Comment (erwanor) [reply pd/src/proxy/tendermint.rs:46]: They're artifacts from the tonic codegen, I'll replace them with more descriptive doc comments!

File: proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto (hunk 3 above)
Comment (erwanor) [reply proto/ibc-go-vendor/cosmos/base/tendermint/v1beta1/query.proto:21]: Agreed! `tendermint/query/node_info` would work. I'm not sure why the Cosmos SDK has different [names](https://docs.cosmos.network/v0.44/core/proto-docs.html#cosmos/base/tendermint/v1beta1/query.proto) for what I think (?) resolves to a [`Status`](https://docs.tendermint.com/master/spec/rpc/#status).

It'd be tempting to just stick to the spec described in the [Tendermint RPC](https://docs.tendermint.com/master/spec/). I'm still getting up-to-speed about the ecosystem, so I'll defer to you on this one

**Comment by hdevalence** [comment]
Hey! As mentioned on Discord, sorry that this got overrun by the backport from Tendermint 0.35 to 0.34.  It seems like doing this well will require some changes to `tendermint-rs` to add in domain types. 

However, right now, we're not in a good place to "close the loop" on a change cycle with the `tendermint-rs`/`ibc-rs` crates, because the way we accomplished the 0.34 backport was to deeply fork the stack of crates depending on `tendermint-rs`/`ibc-rs`. So before we can start making changes and consuming them, we have to work with upstream to have semver releases that we can consume.

For that reason, I think it'd be better to continue to pause this work, to avoid the possibility of effort getting wasted on conflicting changes -- but thanks very much for starting it.

**Pull request closed without merging**